component/prometheus/fanout.go

```diff
@@ -69,11 +78,17 @@ func (f *Fanout) Appender(ctx context.Context) storage.Appender {
 		componentID:  f.componentID,
 		writeLatency: f.writeLatency,
 	}
+
+	updated := false
 	for _, x := range f.children {
 		if x == nil {
 			continue
 		}
 		app.children = append(app.children, x.Appender(ctx))
+		updated = true
+	}
+	if updated {
+		f.samplesCounter.Inc()
```

Comment:
This isn't quite right, this is incrementing for each appender (which is given a batch of samples), per sample. 

See the Append method below; that's the per-sample method.